### PR TITLE
Release process improvements and docs

### PR DESCRIPTION
**Pull request by hashhar**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -355,3 +355,42 @@ See [CHANGELOG](CHANGELOG.md) for details.
...
+  ```
+- Make sure you have a GPG keypair. See
+  https://central.sonatype.org/publish/requirements/gpg/ for more instructions.
+- Since the default branch of this repository cannot be pushed to you need to
+  execute Maven release like so:
+  ```sh
+  ./mvnw release:clean release:prepare -DpushChanges=false
+  ./mvnw release:perform -DlocalCheckout=true
+  ```
+- Once the above finishes you'll have 2 commits locally. Create a new pull
```
Comment (losipiuk) [inline README.md]: Typicallly I just merge those - without a PR

**Review by losipiuk — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (hashhar) [reply README.md]: How? I can't seem to push to the `develop` branch directly.

Comment (losipiuk) [reply README.md]: Maybe it is because I have org admin

Comment (hashhar) [reply README.md]: I'll keep the pull request process here then for the time being until I can think of some solution.

**Comment by hashhar** [comment]
AC. Fixed the build failure using the first two commits.

PTAL @losipiuk

**Review by losipiuk — APPROVED** [review]

**Pull request merged**